### PR TITLE
ci: Add dependabot go-storage auto build support

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

### CODE REVIEW

**Comment by Xuanwo** [comment]
ping @xxchan for a look.

There are the following problems:

- The action won't work until we set go-storage to semver: Dependabot ignore git versions
- I don't know how to test this workflow.

File: .github/workflows/go-storage-auto-build.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: Dependabot go-storage auto build
+on: pull_request_target
+
+permissions:
+  pull-requests: write
+  contents: write
```
Comment (xxchan) [inline .github/workflows/go-storage-auto-build.yml:6]:
```suggestion
on: pull_request
```
is enough?

**Comment by xxchan** [comment]
I think we can test by removing `dependency-names` first?

File: .github/workflows/go-storage-auto-build.yml (hunk 1 above)
Comment (Xuanwo) [reply .github/workflows/go-storage-auto-build.yml:6]: We need `Auto commit & push`, so `write` is required.

**Comment by xxchan** [comment]
Wait, but we have to merge this into master first to test?

**Comment by Xuanwo** [comment]
> Wait, but we have to merge this into master first to test?

Yep, it's a problem. Maybe we can merge this to another branch?

**Comment by Xuanwo** [comment]
Here is an open PR from dependabot: https://github.com/beyondstorage/go-service-s3/pull/138

Maybe we can test via them instead.

**Comment by Xuanwo** [comment]
> > Wait, but we have to merge this into master first to test?
> 
> Yep, it's a problem. Maybe we can merge this to another branch?

I think this is an affordable price: one or two useless commit

**Comment by xxchan** [comment]
> Maybe we can merge this to another branch?

Can it be triggered? I'm OK.

**Comment by Xuanwo** [comment]
image

Ooops, it looks like we have to merge this PR into master first.

**Review by xxchan — APPROVED** [review]

**Pull request merged**